source/components/organisms/Step/Step.tsx

```diff
@@ -37,19 +40,46 @@ interface DialogText {
   title: string;
   body: string;
 }
-const dialogText: Record<DIALOG_TEMPLATE, DialogText> = {
-  [DIALOG_TEMPLATE.MAIN_STEP_COMPLETIONS]: {
-    title: "Vill du stänga formuläret?",
-    body: "Formuläret sparas och du kan fortsätta fylla i det fram till sista dagen för inlämning.",
-  },
-  [DIALOG_TEMPLATE.MAIN_STEP]: {
-    title: "Vill du avbryta ansökan",
-    body: "Ansökan sparas i 3 dagar. Efter det raderas den och du får starta en ny.",
-  },
-  [DIALOG_TEMPLATE.SUB_STEP]: {
-    title: "Vill du stänga fönster utan att spara inmatad uppgift?",
-    body: "",
-  },
+
+const getDialogText = (template: DIALOG_TEMPLATE): DialogText => {
+  switch (template) {
+    case DIALOG_TEMPLATE.MAIN_STEP:
+    case DIALOG_TEMPLATE.NEW_APPLICATION_STEP:
+      return {
+        title: "Vill du avbryta ansökan",
+        body: "Ansökan sparas i 3 dagar. Efter det raderas den och du får starta en ny.",
+      };
+    case DIALOG_TEMPLATE.MAIN_STEP_COMPLETIONS:
+      return {
+        title: "Vill du stänga formuläret?",
+        body: "Formuläret sparas och du kan fortsätta fylla i det fram till sista dagen för inlämning.",
+      };
+    case DIALOG_TEMPLATE.SUB_STEP:
+      return {
+        title: "Vill du stänga fönster utan att spara inmatad uppgift?",
+        body: "",
+      };
+    default:
+      return {
+        title: "",
+        body: "",
+      };
+  }
+};
+
+const getDialogTemplate = (status: string): DIALOG_TEMPLATE => {
```

Comment:
_Might_ be better to move this into a mapping as well, however since it's a lot of many-to-one it might feel verbose. I still prefer it though.

```typescript
const statusTemplateMap: Record<ApplicationStatusType, DIALOG_TEMPLATE> = {
    [ACTIVE_RANDOM_CHECK_REQUIRED_VIVA]: DIALOG_TEMPLATE.MAIN_STEP_COMPLETIONS,
    [ACTIVE_ONGOING_RANDOM_CHECK]: DIALOG_TEMPLATE.MAIN_STEP_COMPLETIONS,
    [ACTIVE_COMPLETION_REQUIRED_VIVA]: DIALOG_TEMPLATE.MAIN_STEP_COMPLETIONS,
    [ACTIVE_ONGOING_COMPLETION]: DIALOG_TEMPLATE.MAIN_STEP_COMPLETIONS,
    [NEW_APPLICATION]: DIALOG_TEMPLATE.NEW_APPLICATION_STEP,
    [ACTIVE_ONGOING_NEW_APPLICATION]: DIALOG_TEMPLATE.NEW_APPLICATION_STEP,
}

function getDialogTemplate(status: string): DIALOG_TEMPLATE {
    return statusTemplateMap[status] ?? DIALOG_TEMPLATE.MAIN_STEP;
}
```